src/showingpreviously/cinemas/arc.py

```diff
@@ -0,0 +1,95 @@
+import re
+import time
+from bs4 import BeautifulSoup
+from datetime import datetime
+
+
+import showingpreviously.requests as requests
+from showingpreviously.model import ChainArchiver, CinemaArchiverException, Chain, Cinema, Film, Showing, Screen
+from showingpreviously.consts import UK_TIMEZONE
+
+
+CINEMAS_INDEX_URL = 'http://arccinema.ie/'
+SHOWING_LIST_URL = '{cinema_url}/?dIndex={date_code}'
+FILM_URL = 'https://helston.merlincinemas.co.uk/ajax/film/{film_id}'
+SCREEN_NAME_PATTERN = re.compile(r'Showing In&nbsp;(?P<screen_name>.+?) -')
+SCREENING_FINISHED_TEXT = 'The performance has expired, or been taken off sale'
+SCREENING_NO_EVENTS_TEXT = 'No event exists for the specified event code.'
+
+CHAIN = Chain('ARC Cinema')
+
+
+def get_response(url: str) -> requests.Response:
+    r = requests.get(url)
+    if r.status_code != 200:
+        raise CinemaArchiverException(f'Got status code {r.status_code} when fetching URL {url}')
+    return r
+
+
+def get_cinemas_as_dict() -> dict[str, Cinema]:
+    r = get_response(CINEMAS_INDEX_URL)
+    soup = BeautifulSoup(r.text, features='html.parser')
+    cinema_list = soup.find('div', {'class': 'links'})
+    cinemas = {}
+    for cinema_link in cinema_list.find_all('a', {'href': True}):
+        cinema_name = cinema_link.text
+        cinema_url = cinema_link['href']
+        cinemas[cinema_url] = Cinema(cinema_name, UK_TIMEZONE)
+    return cinemas
+
+
+def get_date_and_screen(booking_url: str, delay_time: int) -> (datetime, Screen):
+    try:
+        r = get_response(booking_url)
+    except requests.exceptions.ConnectionError:
+        # admit-one.eu has annoying rate limiting. try again, with an increased back-off
+        delay_time += 10
+        time.sleep(delay_time)
+        return get_date_and_screen(booking_url, delay_time)
+    if SCREENING_FINISHED_TEXT in r.text or SCREENING_NO_EVENTS_TEXT in r.text:
```

Comment:
Add an additional check here to prevent screenings from the distant future from being archived.

Reply:
As discussed, no longer doing this in the archivers.